cpp/src/trajectory/trajectory_distances_and_speeds.cu

```diff
@@ -59,12 +59,13 @@ struct dispatch_timestamp {
     auto policy = rmm::exec_policy(stream);
 
     // Compute output column size
-    rmm::device_vector<int32_t> unique_keys(object_id.size());
-    auto last_key_pos =
-        thrust::unique_copy(policy->on(stream), object_id.begin<int32_t>(),
-                            object_id.end<int32_t>(), unique_keys.begin());
-    auto size = thrust::distance(unique_keys.begin(), last_key_pos);
-    unique_keys.~device_vector();
+    auto size = [&]() {
```

Comment:
This is identical with the code in trajectory_bounding_boxes.cu -- make a function?